server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -308,6 +364,145 @@ class DeltaSharedTableKernel(
     throw new DeltaSharingUnsupportedOperationException("not implemented yet")
   }
 
+  private def processBatchByColumnVector(
+      scanFileBatch: FilteredColumnarBatch,
+      version: Long,
+      queryType: QueryTypes.QueryType,
+      respondedFormat: String,
+      table: Table,
+      engine: Engine,
+      isVersionQuery: Boolean,
+      snapshot: SharedTableSnapshot): Seq[Object] = {
+
+    val batchData = scanFileBatch.getData
+    val batchSize = batchData.getSize
+    val addFileVectors = getAddFileColumnVectors(batchData)
+    val selectionVector = scanFileBatch.getSelectionVector
+    var addFileObjects = Seq[Object]()
+    for (rowId <- 0 until batchSize) {
+      val isSelected = !selectionVector.isPresent ||
```

Comment:
@charlenelyu-db 
question: what's the selectionVector? And why do we need to check it?